### ISSUE TITLE
BrentRoots in toolkit.c

### ISSUE THREAD

**Issue by frwagner** [ComputationalThermodynamics/MAGEMin#47]
Hi all:

I installed MAGEMin from source on JUWELS. The code works fine, but crashes if the routine BrentRoots in toolkit.c is called. I think the variable error in toolkit.c line 311 should be the pointer *error.

Could you also have a look into the routine mergeParallelFiles in dump_function.c. I think the code crashes for MPI runs in line 1119, because the file referenced as fp1 could not be opened or found.

I would also like to know for which temperature and pressure range the code can be applied to. Can MAGEMin and its current database from Holland et al., 2018 be used for the entire pressure and temperature range of Earth’s mantle?

Thank you!

Frank

**Comment by NicolasRiel** [comment]
Hi Frank,

Thank you for reporting the issue, this is a new error to me, although we did run the code on a local machine on 64 cores.

I will check the BrentRoots function as soon as possible. 

Concerning the fp1 file, it might be that the directory "output" is not created or added by default to MAGEMin directory. Could you try creating it (same level as for instance "src") to see if that fixes this problem?

The Igneous database from H&P 2018 has been inverted for 700-2000°C and 0-5 GPa, so to lithospheric conditions. 
If you need to work at the mantle scale the, Stixrude (Stix) database is the one you look for.  This database is not available for now in MAGEMin, but a student will work on that starting spring.

Hope this helps,

Nicolas

**Comment by frwagner** [comment]
Dear Nicolas:

Thank you for the clarification regarding the applicable pressure and temperature range of the code.

Coming back to the second issue. I think the directory "output" is created automatically. I think it crashes only for MPI runs using … MAGEMin --Verb=1 …; I suspect that the code will not write _pseudosection_output.i.txt files to output in the first place and the function mergeParallelFiles() will then not work properly. It works e.g. if _pseudosection_output.i.txt files already exist in output from a previous run or if I set gv.verbose in line 60 of dump_function.c to 1.

Cheers,
 Frank

**Comment by frwagner** [comment]
Hi Nicolas:

That is correct, changing error = 0 to pointer *error = 0 works for me.

I have to admit, I did not think about verbose in parallel. I just copied the example from https://computationalthermodynamics.github.io/MAGEMin/ckbk/Commands.html#multiple-points-in-parallel. 

```
mpirun -np 8 ./MAGEMin --Verb=1 --File=/path_to_file/MAGEMin_input.dat --n_points=4 --test=0
mpiexec -n 8 ./MAGEMin --Verb=1 --File=/path_to_file/MAGEMin_input.dat --n_points=4 --test=0
```

This example does not work on my machine, because of the things we discussed. However, using --Verb=0 works again. At least that is my observation.

I also looked at the output provided for MATLAB, but do not see any issues with it. I think these routines work just fine. I used MAGEMin in command-line mode. I did not try the Julia interface or the MATLAB scripts, because I am working at a supercomputing site. I occasionally use Perple_X, but my colleagues from DLR suggested to have a look into your project as possible alternative. I do like your code, because it is much leaner and newer when compared to other options. However, I am interested in deeper depths, thus I am looking forward to databases reaching higher pressures and temperatures.

Cheers,
 Frank

**Comment by NicolasRiel** [comment]
Hi Frank,

Thank you for the feedback. I will change how the different outputs are choosen and update the documentation accordingly.

Concerning  deeper depth have a look at "Stixrude, L., & Lithgow-Bertelloni, C. (2010). Thermodynamics of the Earth’s mantle. Reviews in Mineralogy and Geochemistry, 71(1), 465-484" and reference therein. There was an update last year, and this is the one we will implement.

Cheers

**Comment by NicolasRiel** [comment]
The BrentRoots pointer mistake is  now fixed in version 1.2.8.
I also updated the documentation to prevent having the crash for MPI runs using … MAGEMin --Verb=1

I close the issue,

**Comment by frwagner** [comment]
I can confirm that the fixed version 1.2.8 solves the issue. Thank you!
